### PR TITLE
Add .NET Core builds

### PR DESCRIPTION
**Pull request by chwarr**
Note to reviewer/merger: please perform a full merge commit for this.

GitHub is bad about tracking history of code and comments when rebasing. Since this PR consists of multiple commits we want to merge, I'm manually tracking the "iterations" of the PR in branches in my fork.

# Iterations #
* [1](https://github.com/chwarr/bond/tree/chwarr/dnc/pr/1): 9749deb - initial
* [2](https://github.com/chwarr/bond/tree/chwarr/dnc/pr/2): e2f34f8 - "DNC" to "C# .NET Core", spelling
* [3](https://github.com/chwarr/bond/tree/chwarr/dnc/pr/3): 29ea9e2 - CHANGELOG
* [4](https://github.com/chwarr/bond/tree/chwarr/dnc/pr/4): d2b5504 - copy edit CHANGELOG

### CODE REVIEW

**Comment by chwarr** [comment]
Closes https://github.com/Microsoft/bond/issues/165
Closes https://github.com/Microsoft/bond/issues/166

File: appveyor.yml [hunk 1]
```diff
@@ -43,6 +43,8 @@
             - BOND_BUILD: C#
               BOND_OUTPUT: Fields
               BOND_CONFIG: Fields
+            - BOND_BUILD: DNC
```
Comment (tstein) [inline appveyor.yml]: C#-DNC or C#-dotnetcore? This seems needlessly opaque.

File: cs/dnc/build.ps1 [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+try
+{
+    $script:msb_logger = $null
+    if ($MSBuildLogger) {
+        $script:msb_logger = "/logger:$($MSBuildLogger)"
+    }
+
+    # We push to a stack other than the default one so that we don't
+    # accidenally change its contents.
```
Comment (tstein) [inline cs/dnc/build.ps1]: acciden*t*ally

File: cs/src/io/unsafe/InputPointer.cs [hunk 3]
```diff
@@ -233,7 +233,11 @@ public string ReadString(Encoding encoding, int size)
                 EndOfStream(size);
             }
 
-            var result = new string((sbyte*) buffer, position, size, encoding);
+#if !(NETSTANDARD1_3 || NETSTANDARD1_6)
+            string result = new string((sbyte*) buffer, position, size, encoding);
+#else
+            string result = encoding.GetString(buffer + position, size);
```
Comment (chwarr) [inline cs/src/io/unsafe/InputPointer.cs]: Can we just use the `encoding.GetString()` version everywhere?

Comment (chwarr) [inline cs/src/io/unsafe/InputPointer.cs]: No, we cannot. `Encoding.GetString(Byte*, Int32)` was added in .NET 4.6.

File: appveyor.yml (hunk 1 above)
Comment (chwarr) [inline appveyor.yml]: Changed to "C# .NET Core"

File: cs/dnc/build.ps1 (hunk 2 above)
Comment (chwarr) [inline cs/dnc/build.ps1]: Fixed. Thanks!

**Comment by chwarr** [comment]
Merged in commit f80a8b5baec43a74960b23c5ad0fc628118f6e6f. Closing.

**Pull request closed without merging**